cruise-control/src/main/java/com/linkedin/kafka/cruisecontrol/servlet/response/ClusterPartitionState.java

```diff
@@ -112,25 +115,26 @@ protected void populateKafkaPartitionState(Set<PartitionInfo> underReplicatedPar
         }
         for (PartitionInfo partitionInfo : _kafkaCluster.partitionsForTopic(topic)) {
           int numInsyncReplicas = partitionInfo.inSyncReplicas().length;
+          boolean hasOfflineReplicas = partitionInfo.offlineReplicas().length != 0;
           boolean isURP = numInsyncReplicas != partitionInfo.replicas().length;
+          boolean isOffline = partitionInfo.leader() == null;
```

Comment:
Does this correctly reflect the leader is not alive? E.g. in kafka_cluster_state, it shows the leader is [-1]. Could you help to check why that is -1 while here it is null?

Reply:
Good question - I believe it does. In `PartitionState.java` [(here)](https://github.com/linkedin/cruise-control/blob/migrate_to_kafka_2_5/cruise-control/src/main/java/com/linkedin/kafka/cruisecontrol/servlet/response/PartitionState.java#L54), if the partition leader is null, `_leader` is set to `-1`. It looks like we use `PartitionState` to populate the kafkaClusterState [here](https://github.com/linkedin/cruise-control/blob/migrate_to_kafka_2_5/cruise-control/src/main/resources/yaml/responses/kafkaClusterState.yaml#L121-L148).

The null leader status is used to determine offline partitions elsewhere in the code as well [(example)](https://github.com/linkedin/cruise-control/blob/migrate_to_kafka_2_5/cruise-control/src/main/java/com/linkedin/kafka/cruisecontrol/monitor/MonitorUtils.java#L451-L452).

Reply:
Perfect. Thank you!